packages/web/src/shared/components/TextInput.tsx

```diff
@@ -1,16 +1,32 @@
 import { forwardRef, ComponentProps } from 'react';
+import { useFormContext } from 'react-hook-form';
+import useMergedRef from '../mergeRefs';
+
+export interface TextInputProps extends Omit<ComponentProps<'input'>, 'name'> {
+  name: string;
+}
+
+const TextInput = forwardRef<HTMLInputElement, TextInputProps>(
+  ({ className = '', name, onChange, onBlur, required, ...props }, ref) => {
+    const context = useFormContext();
+    const hasErrors = !!context?.formState.errors[name];
+    const registerProps = context?.register(name, {
+      required,
+      onChange,
+      onBlur,
+    });
```

Comment:
Now an input component can use the form helpers to register the input in the form. This allows the form to track whether the component has been touched, is dirty, or is valid